script.js

```diff
@@ -1,4 +1,344 @@
+
+    switch (programState.gameType) {
+      case GameType.ScissorsPaperStone:
+        {
+          spsGame(rawInput);
+        }
+        break;
+      case GameType.GuessTheWord:
+        {
+          guessTheWord(rawInput);
+        }
+        break;
+      case GameType.DiceGame:
+        {
+          diceGame(rawInput);
+        }
+        break;
+    }
+  }
+}
+
+function diceGame(rawInput) {
+  const diceRoll = Math.floor(Math.random() * 6) + 1;
+  if (Number(rawInput) === diceRoll) {
+    displayResults(`You've guessed ${Number(rawInput)}, and the roll was ${diceRoll}! You win!`);
+  } else {
+    displayResults(`Sorry, no dice! The roll was ${diceRoll}.`);
+  }
+}
+
+function guessTheWord(rawInput) {
+  if (rawInput === "palatable papaya") {
+    displayResults("You've guessed the word correctly! You win!");
+  } else {
+    displayResults("Sorry, you guessed the word incorrectly!");
+  }
+}
+
+function spsGame(rawInput) {
+  switch (rawInput) {
+    // SPS modes
+    case "reversed":
+      {
+        if (programState.gameMode !== GameMode.Reversed) {
+          programState.gameMode = GameMode.Reversed;
+          document.querySelector("#input-sps").value = "";
+          instructionsText.innerHTML += `Reversed mode activated! Enter scissors, paper or stone (enter random for a random choice):`;
+        }
+      }
+      break;
+    case "korean":
+      {
+        if (programState.gameMode !== GameMode.Korean) {
+          programState.gameMode = GameMode.Korean;
+          document.querySelector("#input-sps").value = "";
+          instructionsText.innerHTML += `Korean mode activated! Enter scissors, paper or stone (enter random for a random choice):`;
+        }
+      }
+      break;
+    case "random":
+      {
+        const randomIndex = Math.floor(Math.random() * Object.keys(Choices).length);
+        const randomChoice = Object.values(Choices)[randomIndex];
+        runGame(randomChoice);
+      }
+      break;
+    default:
+      runGame(rawInput);
+      break;
+  }
+}
+
+function runGame(rawInput) {
+  const randomIndex = Math.floor(Math.random() * Object.keys(Choices).length);
+  const computerChoice = Object.values(Choices)[randomIndex];
+
+  let playerInput = null;
+  let winState = null;
+
+  Object.values(Choices).forEach((value) => {
+    if (rawInput.includes(value)) {
+      playerInput = value;
+    }
+  });
+
+  if (playerInput === null) {
+    displayResults("Please enter a valid input");
+  } else {
+    winState = getWinState(playerInput, computerChoice, programState.gameMode);
+    if (programState.gameMode != GameMode.Korean) {
+      calculateScore(playerInput, computerChoice, winState);
+    }
+  }
+}
+
+function getWinState(playerInput, computerChoice, gameMode) {
+  let winState = null;
+  // Check default win state
+  switch (playerInput) {
+    case Choices.Scissors:
+      {
+        if (computerChoice === Choices.Scissors) {
```

Comment:
All draws conditions can be simplified as
if ( computerChoices === playerInput)